client/tileset/tilespec.cpp

```diff
@@ -5450,6 +5451,11 @@ void fill_basic_city_sprite_array(const struct tileset *t,
   if (pcity->client.unhappy) {
     ADD_SPRITE_FULL(t->sprites.city.disorder);
   }
+  if (t->sprites.city.happy != nullptr) {
+    if (pcity->client.unhappy) {
+      ADD_SPRITE_FULL(t->sprites.city.disorder);
+    }
+  }
```

Comment:
Two changes:
* Untested fix to use the happy sprite when the city is happy instead of showing the disorder sprite twice
* Reformat the `if` clauses a bit
```suggestion
  } else if (t->sprites.city.happy != nullptr && pcity->client.happy) {
    ADD_SPRITE_FULL(t->sprites.city.happy);
  }
```

Reply:
LOL. I pushed the wrong commit. Brb.